Assets/SEE/Layout/NodeLayouts/IncrementalTreeMap/LocalMove.cs

```diff
@@ -0,0 +1,423 @@
+        }
+
+        /// <summary>
+        /// applies the local move for the case that the nodes are separated vertical
+        /// </summary>
+        /// <param name="leftNode">the node on the <see cref="Direction.Left"/> side</param>
+        /// <param name="rightNode">the node on the <see cref="Direction.Right"/> side</param>
+        private void apply_flipOnVerticalSegment(Node leftNode, Node rightNode)
+        {
+            // clockwise            anticlockwise
+            // [l][r] -> [lll]      [l][r] -> [rrr]
+            // [l][r]    [rrr]      [l][r]    [lll]
+
+            // adjust rectangles
+            double width = leftNode.Rectangle.Width + rightNode.Rectangle.Width;
+            double ratio = leftNode.Rectangle.Area() / (leftNode.Rectangle.Area() + rightNode.Rectangle.Area());
+
+            leftNode.Rectangle.Width = width;
+            rightNode.Rectangle.Width = width;
+            rightNode.Rectangle.X = leftNode.Rectangle.X;
+
+            leftNode.Rectangle.Depth *= ratio;
+            rightNode.Rectangle.Depth *= (1 - ratio);
+            if (_clockwise)
+            {
+                leftNode.Rectangle.Z = rightNode.Rectangle.Z + rightNode.Rectangle.Depth;
+            }
+            else
+            {
+                rightNode.Rectangle.Z = leftNode.Rectangle.Z + leftNode.Rectangle.Depth;
+            }
+
+            // switch segments
+            Segment newRightSegmentForLeftNode = rightNode.SegmentsDictionary()[Direction.Right];
+            Segment newLeftSegmentForRightNode = leftNode.SegmentsDictionary()[Direction.Left];
+            Segment middle = leftNode.SegmentsDictionary()[Direction.Right];
+            middle.IsVertical = false;
+
+            leftNode.RegisterSegment(newRightSegmentForLeftNode, Direction.Right);
+            rightNode.RegisterSegment(newLeftSegmentForRightNode, Direction.Left);
+
+            if (_clockwise)
+            {
+                leftNode.RegisterSegment(middle, Direction.Lower);
+                rightNode.RegisterSegment(middle, Direction.Upper);
+            }
+            else
+            {
+                leftNode.RegisterSegment(middle, Direction.Upper);
+                rightNode.RegisterSegment(middle, Direction.Lower);
+            }
+        }
+
+        /// <summary>
+        /// applies the local move for the case that the nodes are separated horizontal
+        /// </summary>
+        /// <param name="lowerNode">the node on the <see cref="Direction.Lower"/> side</param>
+        /// <param name="upperNode">the node on the <see cref="Direction.Upper"/> side</param>
+        private void apply_flipOnHorizontalSegment(Node lowerNode, Node upperNode)
+        {
+            // clockwise                anticlockwise
+            // [uuu]  ->  [l][u]        [uuu]  ->  [u][l]
+            // [lll]      [l][u]        [lll]      [u][l]
+
+            // adjust rectangles
+            double depth = lowerNode.Rectangle.Depth + upperNode.Rectangle.Depth;
```

Comment:
Yes, i thought about that too and the existing code is the best solution i found, since its readable but also every method generalizes two different cases in the first part.
Im open for a better compromise between redundancy and readability, but i have no suggestion myself.

Reply:
In that case, I think we should wait for @koschke to take a look at this.